File: UtilidadesDB_DataAccess/Models/ReportsHistory.cs
using System;
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations;

namespace UtilidadesDB_DataAccess.Models
{
    public partial class ReportsHistory
    {
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public int Id { get; set; }
        [Required]
        public DateTime Date { get; set; }
        [Required]
        public string User { get; set; }
        [Required]
        public string ReportType { get; set; }
        [Required]
        public string Json { get; set; }
        public decimal? NewCompanies { get; set; }
        public decimal? NewMoney { get; set; }
    }
}

File: Frontend/ClientApp/src/app/models/Reports/reports-history.ts
export interface ReportsHistory {
  id?: number
  date: Date;
  user: string;
  report_type: string;
  reportTypeNumber: number;
  json: string;
  new_companies?: number;
  new_money?: number;
}

export interface ReportsHistory_JsonNE {
  jsonOPP: string;
  jsonPROJECT: string;
}

export interface ReportsHistory_JsonFE {
  jsonOPP: string;
  jsonPROJECT: string;
}

export interface ReportsHistory_JsonDS {
  jsonOPP: string;
  jsonPROJECT: string;
}

export interface ReportsHistory_JsonRE {
  jsonOPP: string;
  jsonPROJECT: string;
}

export interface ReportsHistory_JsonDE {
  jsonOPP: string;
  jsonPROJECT: string;
}


